DyalogBuild.dyalog

```diff
@@ -312,10 +316,13 @@
                       :Select lc 3⊃⎕NPARTS fl
                       :CaseList '.dyalog' '.aplc' '.aplf' '.apln' '.aplo' '.apli'
                           :Trap DEBUG↓0 ⍝↓↓↓↓ be sure to pass target as a ref, bad things may happen otherwise ()
-                              res←0({0::⍵ ⋄ ⍎⍵}target)0 ⎕SE.SALT.Load fl,' ',options
-     
-                              ⍝source←⎕SE.SALT.Load fl,' -source=no'  ⍝ unfortunately need two calls to establish function & get source
-                              ⍝source←1⊃⎕NGET fl
+                              :If {(1=≢⍵)∧⊃1↑,⍵}~(⎕NC⍕target)∊0 9  ⍝ deal with name clashes
+                              :AndIf (,'#')≢,⍕target
+                                  ('target="',(⍕target),'" exists already with ⎕NC=',(⍕⎕NC target),' and is protected')⎕SIGNAL(∨/' -protect'⍷options)/11
+                                  ⎕EX target
+                              :EndIf
+                              ref←{0::⍵ ⋄ ⍎⍵}target
```

Comment:
This seems a little presumptuous - on any error return the argument?

Reply:
Reformulated that statement.